wp/lib/bap_wp/src/environment.ml

```diff
@@ -177,44 +182,33 @@ let wp_rec_call :
   (t -> Constr.t -> start:Graphs.Ir.Node.t -> Graphs.Ir.t -> t) ref =
   ref (fun _ _ ~start:_ _ -> assert false)
 
+let loop_invariant_checker_rec_call : (string -> loop_handler) ref =
+  ref (fun _ -> assert false)
+
 let trivial_constr (env : t) : Constr.t =
   get_context env
   |> Z3.Boolean.mk_true
   |> Constr.mk_goal "true"
   |> Constr.mk_constr
 
-(* Looks up the precondition of the exit node of a loop by:
+(* Looks up the exit node of a loop by:
    - obtaining the post dominator tree
    - for each node in the SCC, find its parent in the dominator tree
    - if the parent node is not in the original SCC, it is an exit node *)
-let loop_exit_pre (env : t) (node : Graphs.Ir.Node.t) (graph : Graphs.Ir.t)
-  : Constr.t option =
+let loop_exit (node : Graphs.Ir.Node.t) (graph : Graphs.Ir.t)
```

Comment:
It might be nice to summarize these kinds of changes in the PR request, to help my tiny brain wrap around them.

Reply:
I also wonder if the comments couldn't use some upgrades.

Reply:
I can update the comments. This function originally found the exit node and returned its precondition. I changed it to return the exit node itself, so the handler can look up the precondition after visiting the subgraph.